test/Elastic.Apm.Tests/MetricsTests.cs

```diff
@@ -0,0 +1,128 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Elastic.Apm.Api;
+using Elastic.Apm.Logging;
+using Elastic.Apm.Metrics;
+using Elastic.Apm.Metrics.MetricsProvider;
+using Elastic.Apm.Tests.Mocks;
+using FluentAssertions;
+using Xunit;
+
+namespace Elastic.Apm.Tests
+{
+	public class MetricsTests
+	{
+		[Fact]
+		public void CollectAllMetrics()
+		{
+			var mockPayloadSender = new MockPayloadSender();
+			var testLogger = new TestLogger();
+			var mc = new MetricsCollector(testLogger, mockPayloadSender, new TestAgentConfigurationReader(testLogger));
+
+			mc.CollectAllMetrics();
+
+			mockPayloadSender.Metrics.Should().NotBeEmpty();
+		}
+
+		[Fact]
+		public void SystemCpu()
+		{
+			if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux) && !RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) return;
+
+			var systemTotalCpuProvider = new SystemTotalCpuProvider();
+
+			//Some implementation may need to be called at least 2 times to deliver value - this is by design
```

Comment:
It's not the case anymore and `systemTotalCpuProvider.GetSamples();` below is not necessary, right?

Reply:
Yes, removed, thanks!